### PR TITLE
fix: missing UI connections in console

### LINKED ISSUE #6458: missing UI connection from Api to Bucket
### I tried this:

```wing
bring cloud;

let api = new cloud.Api();
let b = new cloud.Bucket();
api.post("/doc/:id", inflight (req) => {
  b.put(req.vars.get("id"), req.body! );
});

```

### This happened:

image

### I expected this:

To see connection between api gateway to bucket

### Is there a workaround?

_No response_

### Anything else?

_No response_

### Wing Version

0.73.41

### Node.js Version

_No response_

### Platform(s)

_No response_

### Community Notes

* Please vote by adding a 👍 reaction to the issue to help us prioritize.
* If you are interested to work on this issue, please leave a comment.

### PR DESCRIPTION
**Pull request by Chriscbr**
Fixes #6458

Some UI connections were missing in the Wing Console because when generating information about different resource nodes, inflight closures created in Winglang were treated as resources while inflight closures created using the TypeScript framework were not. To fix this discrepancy, this PR changes how connections are generated so that all intermediate inflight closures are no longer modeled as nodes. Instead, inflight closures (both those created in Winglang and in TypeScript) are explored until resource-like nodes are reached.

Taking this app as an example:

```js
let b = new cloud.Bucket();

let listData = inflight () => {
    b.list();
};

new cloud.Function(inflight () => {
    listData();
    b.put("hello.txt", "world");
});
```

Previously the SDK generated this connection data:

```js
{
  "source": "root/Default/$Closure1_0",
  "sourceOp": "handle",
  "target": "root/Default/Bucket",
  "targetOp": "list",
  "name": "list"
},
{
  "source": "root/Default/$Closure2_0",
  "sourceOp": "handle",
  "target": "root/Default/Bucket",
  "targetOp": "put",
  "name": "put"
},
{
  "source": "root/Default/$Closure2_0",
  "sourceOp": "handle",
  "target": "root/Default/$Closure1_0",
  "targetOp": "handle",
  "name": "handle"
},
{
  "source": "root/Default/Function",
  "sourceOp": "invoke",
  "target": "root/Default/$Closure2_0",
  "targetOp": "handle",
  "name": "handle"
},
{
  "source": "root/Default/Function",
  "sourceOp": "invokeAsync",
  "target": "root/Default/$Closure2_0",
  "targetOp": "handle",
  "name": "handle"
}
```

Now it generates this connection data (note that all "closure" nodes are gone - we just emit relationships between real resources):

```js
{
  "source": "root/Default/Function",
  "sourceOp": "invoke",
  "target": "root/Default/Bucket",
  "targetOp": "put",
  "name": "put"
},
{
  "source": "root/Default/Function",
  "sourceOp": "invoke",
  "target": "root/Default/Bucket",
  "targetOp": "list",
  "name": "list"
},
{
  "source": "root/Default/Function",
  "sourceOp": "invokeAsync",
  "target": "root/Default/Bucket",
  "targetOp": "put",
  "name": "put"
},
{
  "source": "root/Default/Function",
  "sourceOp": "invokeAsync",
  "target": "root/Default/Bucket",
  "targetOp": "list",
  "name": "list"
}
```

## Checklist

- [x] Title matches [Winglang's style guide](https://www.winglang.io/contributing/start-here/pull_requests#how-are-pull-request-titles-formatted)
- [x] Description explains motivation and solution
- [x] Tests added (always)
- [ ] Docs updated (only required for features)
- [ ] Added `pr/e2e-full` label if this feature requires end-to-end testing

*By submitting this pull request, I confirm that my contribution is made under the terms of the [Wing Cloud Contribution License](https://github.com/winglang/wing/blob/main/CONTRIBUTION_LICENSE.md)*.

### CODE REVIEW

**Comment by Chriscbr** [comment]
The map for the code in #6458 now looks like:

Screenshot 2024-05-16 at 5 51 25 PM

File: libs/wingsdk/src/std/resource.ts [hunk 1]
```diff
@@ -193,26 +197,33 @@ export abstract class Resource extends Construct implements IResource {
...
+            source: construct,
+            sourceOp: baseOp ?? op,
             target: dep,
             targetOp: depOp,
             name: depOp,
           });
         }
+      } else if (hasLiftMap(dep)) {
+        // case 2: dep is an inflight
+        addConnectionsFromLiftMap(construct, dep._liftMap!, baseOp ?? op);
```
Comment (MarkMcCulloh) [inline libs/wingsdk/src/std/resource.ts:226]: Did the type guard on `hasLiftMap` not work?

```suggestion
        addConnectionsFromLiftMap(construct, dep._liftMap, baseOp ?? op);
```

File: libs/wingsdk/src/target-sim/api.ts [hunk 2]
```diff
@@ -41,6 +41,17 @@ export class Api extends cloud.Api implements ISimulatorResource {
...
+  /** @internal */
+  public get _liftMap(): LiftMap {
+    const lifts: Array<[IResource, Array<string>]> = [];
+    for (const handler of Object.values(this.handlers)) {
+      lifts.push([handler.func, [cloud.FunctionInflightMethods.INVOKE]]);
+    }
+    return {
+      [cloud.ApiInflightMethods.REQUEST]: lifts,
+    };
+  }
```
Comment (MarkMcCulloh) [inline libs/wingsdk/src/target-sim/api.ts:53]: What happens if this isn't here? `addEndpoint` Seems to take care of adding a connection between this resource and the underlying function

**Review by MarkMcCulloh — APPROVED** [review]

File: libs/wingsdk/src/std/resource.ts (hunk 1 above)
Comment (Chriscbr) [reply libs/wingsdk/src/std/resource.ts]: Fixed

File: libs/wingsdk/src/target-sim/api.ts (hunk 2 above)
Comment (Chriscbr) [reply libs/wingsdk/src/target-sim/api.ts:53]: Mmmm - yeah I guess there's some double duty being performed. I tried this and the connection still appears correctly when I test this out in Wing Console locally 👍  so removing for now

**Pull request merged**